src/main.ts

```diff
@@ -586,8 +587,17 @@ function makeTimestampMethods(options: Options, longs: ReturnType<typeof makeLon
 }
 
 // When useOptionals=true, non-scalar fields are translated into optional properties.
-function isOptionalProperty(field: FieldDescriptorProto, options: Options): boolean {
-  return (options.useOptionals && isMessage(field) && !isRepeated(field)) || field.proto3Optional;
+// When forceOptionals=true, all fields are translated into optional properties, except map Entry key/values.
+function isOptionalProperty(
+  field: FieldDescriptorProto,
+  messageOptions: MessageOptions | undefined,
+  options: Options
+): boolean {
+  return (
+    (options.useOptionals && isMessage(field) && !isRepeated(field)) ||
+    (options.forceOptionals && !messageOptions?.mapEntry) ||
```

Comment:
Per discussion, I'm good with removing `options.forceOptionals` and just remove the `isMessage` / `!isRepeated` from the existing check.

Also I assume `!...mapEntry` is in here just b/c there are some nuances to making it optional that you didn't want to tackle in this PR? If so, maybe just add an in-source comment/todo that it's for later.

Reply:
Yes, otherwise I had to assert that the Map entry key is set in various places, which seems silly as the Map entry key is mandatory.